integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -0,0 +1,377 @@
+            hnsw_index_creation_kwargs=self.hnsw_index_creation_kwargs,
+            hnsw_ef_search=self.hnsw_ef_search,
+        )
+
+    def _execute_sql(
+        self, sql_query: Query, params: Optional[tuple] = None, error_msg: str = "", cursor: Optional[Cursor] = None
+    ):
+        """
+        Internal method to execute SQL statements and handle exceptions.
+
+        :param sql_query: The SQL query to execute.
+        :param params: The parameters to pass to the SQL query.
+        :param error_msg: The error message to use if an exception is raised.
+        :param cursor: The cursor to use to execute the SQL query. Defaults to self._cursor.
+        """
+
+        params = params or ()
+        cursor = cursor or self._cursor
+
+        try:
+            result = cursor.execute(sql_query, params)
+        except Error as e:
+            self._connection.rollback()
+            raise DocumentStoreError(error_msg) from e
+        return result
+
+    def _create_table_if_not_exists(self):
+        """
+        Creates the table to store Haystack documents if it doesn't exist yet.
+        """
+
+        table_structure_str = ", ".join(f"{name} {dtype} {constraint}" for name, dtype, constraint in TABLE_DEFINITION)
+
+        create_sql = SQL("CREATE TABLE IF NOT EXISTS {table_name} (" + table_structure_str + ")").format(
+            table_name=Identifier(self.table_name), embedding_dimension=SQLLiteral(self.embedding_dimension)
+        )
+
+        self._execute_sql(create_sql, error_msg="Could not create table in PgvectorDocumentStore")
+
+    def delete_table(self):
+        """
+        Deletes the table used to store Haystack documents.
+        """
+
+        delete_sql = SQL("DROP TABLE IF EXISTS {table_name}").format(table_name=Identifier(self.table_name))
+
+        self._execute_sql(delete_sql, error_msg="Could not delete table in PgvectorDocumentStore")
+
+    def _handle_hnsw(self):
+        """
+        Internal method to handle the HNSW index creation.
+        It also sets the hnsw.ef_search parameter for queries if it is specified.
+        """
+
+        if self.hnsw_ef_search:
+            sql_set_hnsw_ef_search = SQL("SET hnsw.ef_search = {hnsw_ef_search}").format(
+                hnsw_ef_search=SQLLiteral(self.hnsw_ef_search)
+            )
+            self._execute_sql(sql_set_hnsw_ef_search, error_msg="Could not set hnsw.ef_search")
+
+        index_esists = bool(
+            self._execute_sql(
+                "SELECT 1 FROM pg_indexes WHERE tablename = %s AND indexname = %s",
+                (self.table_name, HNSW_INDEX_NAME),
+                "Could not check if HNSW index exists",
+            ).fetchone()
+        )
+
+        if index_esists and not self.hnsw_recreate_index_if_exists:
+            logger.warning(
+                "HNSW index already exists and won't be recreated. "
+                "If you want to recreate it, set hnsw_recreate_index=True"
```

Comment:
"If you want to recreate it, pass 'hnsw_recreate_index_if_exists=True' to the document store constructor" 

or something like that...